packages/matter-node.js/src/Device.ts

```diff
@@ -163,28 +170,35 @@ class Device {
                     )
                 ])
                 .addEndpoint(0x01, DEVICE.ON_OFF_LIGHT, [onOffClusterServer])
-            )
-            .start()
+            );
+        device.start()
 
         logger.info("Listening");
-
-        const qrPairingCode = QrPairingCodeCodec.encode({
-            version: 0,
-            vendorId: vendorId.id,
-            productId,
-            flowType: CommissionningFlowType.Standard,
-            discriminator,
-            passcode,
-            discoveryCapabilities: DiscoveryCapabilitiesSchema.encode({
-                ble: false,
-                softAccessPoint: false,
-                onIpNetwork: true,
-            }),
-        });
-        console.log(QrCode.encode(qrPairingCode));
-        console.log(`QR Code URL: https://project-chip.github.io/connectedhomeip/qrcode.html?data=${qrPairingCode}`);
-        console.log(`Manual pairing code: ${ManualPairingCodeCodec.encode({ discriminator, passcode })}`);
+        if (!device.isCommissioned()) {
+            const qrPairingCode = QrPairingCodeCodec.encode({
+                version: 0,
+                vendorId: vendorId.id,
+                productId,
+                flowType: CommissionningFlowType.Standard,
+                discriminator,
+                passcode,
+                discoveryCapabilities: DiscoveryCapabilitiesSchema.encode({
+                    ble: false,
+                    softAccessPoint: false,
+                    onIpNetwork: true,
+                }),
+            });
+            console.log(QrCode.encode(qrPairingCode));
+            console.log(`QR Code URL: https://project-chip.github.io/connectedhomeip/qrcode.html?data=${qrPairingCode}`);
+            console.log(`Manual pairing code: ${ManualPairingCodeCodec.encode({ discriminator, passcode })}`);
+        } else {
+            console.log("Device is already commissioned. Waiting for controllers to connect ...");
+        }
     }
 }
 
 new Device().start().catch(error => logger.error(error));
+
+process.on("SIGINT", () => {
```

Comment:
It would be nice to hide this outside of the app.

Reply:
Do you have an example how? Right now users start Device "file" and end with CTRL-C basically ... it was the only way I found (at least with the current structure) how to asure that the data are stored.